File: src/stan-users-guide/simple_linear_regression/regression_1.py
""" Example program that generates data, compiles and runs regression.stan"""
from cmdstanpy import CmdStanModel
import numpy


# Run from command line: Python regression_1.py

alpha = 2
beta = 3
sigma = 5
n = 100
x = numpy.random.uniform(size=n)
y = numpy.random.normal(size=n, loc=alpha + beta * x, scale=sigma)

stan_data = {'N': n, 'x': x, 'y': y}

stan_program = CmdStanModel(stan_file='regression_1.stan')
stan_program.compile()
fit = stan_program.sample(data=stan_data, output_dir='.')
print(fit.summary())

File: src/stan-users-guide/simple_linear_regression/regression_1.R
library("cmdstanr")

#Run from command line: Rscript regression_1.R
#If running from RStudio remember to set the working directory 
# >Session>Set Working Directory>To Source File Location

#generate data
alpha <- 2
beta <- 3
sigma <- 5
N <- 100
x <- runif(N, 0, 10)
y <- rnorm(N, alpha + beta*x, sigma)
stan_data <- list(N=N, x=x, y=y)

model = cmdstan_model('regression_1.stan')

fit <- model$sample(data = stan_data,num_chains=4)

print(fit$summary())


